src/ui/composables/alignmentContext.js

```diff
@@ -0,0 +1,131 @@
+    if (!keys.length) {
+      return;
+    }
+    return (
+      path
+        //get just the identifiers
+        .map((domainObject) => {
+          return openmct.objects.makeKeyString(domainObject.identifier);
+        })
+        .reverse()
+        // find a match
+        .find((keyString) => {
+          return keys.includes(keyString);
+        })
+    );
+  };
+
+  // Use the furthest ancestor's alignment if it exists, otherwise, use your own
+  let alignmentKey = getAlignmentKeyForPath(path);
+
+  if (!alignmentKey) {
+    const targetObjectKey = openmct.objects.makeKeyString(targetObject.identifier);
+    alignmentKey = targetObjectKey;
+    alignmentMap[targetObjectKey] = ref({
+      leftWidth: 0,
+      rightWidth: 0,
+      multiple: false,
+      axes: {}
+    });
+  }
+
+  // reset any alignment data for the given key
+  const resetAlignment = () => {
+    const key = getAlignmentKeyForPath(path);
+    if (key && alignmentMap[key]) {
+      delete alignmentMap[key];
+    }
+  };
+
+  // Given the axes ids and widths, calculate the max left and right widths and whether or not multiple left axes exist
+  const processAlignment = () => {
+    const axesKeys = Object.keys(alignmentMap[alignmentKey].value.axes);
+    const leftAxes = axesKeys.filter((axis) => axis <= 2);
+    const rightAxes = axesKeys.filter((axis) => axis > 2);
+    // Get width of left Y axis
+    let leftWidth = 0;
+    leftAxes.forEach((leftAxis) => {
+      leftWidth = leftWidth + alignmentMap[alignmentKey].value.axes[leftAxis];
+    });
+    alignmentMap[alignmentKey].value.leftWidth = leftWidth;
+
+    // Get width of right Y axis
+    let rightWidth = 0;
+    rightAxes.forEach((rightAxis) => {
+      rightWidth = rightWidth + alignmentMap[alignmentKey].value.axes[rightAxis];
+    });
+    alignmentMap[alignmentKey].value.rightWidth = rightWidth;
+
+    alignmentMap[alignmentKey].value.multiple = leftAxes.length > 1;
+  };
+
+  /**
+   * Unregister y-axis from width calculations
+   * @param {Object: yAxisId, updateObjectPath} the last known width of the y-axis, yAxisId of the tick bar, path of the axis being updated
+   */
+  const remove = ({ yAxisId, updateObjectPath, type } = {}) => {
+    const key = getAlignmentKeyForPath(updateObjectPath);
+    if (key) {
+      if (alignmentMap[alignmentKey].value.axes[yAxisId] !== undefined) {
+        delete alignmentMap[alignmentKey].value.axes[yAxisId];
+      }
+      processAlignment();
+    }
+  };
+
+  /**
+   * Update widths of a y axis given the id and path. The path is used to determine which ancestor should hold the alignment
+   * @param {Object: width, yAxisId, updateObjectPath} the width of the y-axis, yAxisId of the tick bar, path of the axis being updated
+   */
+  const update = ({ width, yAxisId, updateObjectPath, type } = {}) => {
+    const key = getAlignmentKeyForPath(updateObjectPath);
+    if (key) {
+      if (alignmentMap[alignmentKey].value.axes[yAxisId] === undefined) {
```

Comment:
We could use some conditional chaining to ensure this doesn't throw if it ends up undefined for any reason along the way:
```suggestion
      if (alignmentMap[alignmentKey].value?.axes?.[yAxisId] === undefined) {
```